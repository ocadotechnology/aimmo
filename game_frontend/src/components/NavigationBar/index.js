import React, { Component } from 'react'
import styled from 'styled-components'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import IconButton from '@material-ui/core/IconButton'
import CloseIcon from 'components/icons/Close'

export const NavigationBarLayout = styled.nav`
    grid-area: navigation-bar;
`

export const CloseToolbar = styled(Toolbar)`
  justify-content: flex-end;
`

export default class NavigationBar extends Component {
  render () {
    return (
      <NavigationBarLayout>
        <AppBar
          color='secondary'
          position='sticky'>
          <CloseToolbar>
            <IconButton
              href='/aimmo'
              aria-label='Close'
              color='inherit'>
              <CloseIcon />
            </IconButton>
          </CloseToolbar>
        </AppBar>
      </NavigationBarLayout>
    )
  }
}
