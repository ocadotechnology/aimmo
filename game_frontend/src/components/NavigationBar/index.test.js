/* eslint-env jest */
import React from 'react'
import NavigationBar, { NavigationBarLayout } from 'components/NavigationBar'
import createShallowWithTheme from 'testHelpers/createShallow'

describe('<NavigationBar />', () => {
  it('renders correctly', () => {
    const tree = createShallowWithTheme(<NavigationBar />)
    expect(tree).toMatchSnapshot()
  })
})

describe('<NavigationBarLayout />', () => {
  it('renders correctly', () => {
    const tree = createShallowWithTheme(<NavigationBarLayout />)
    expect(tree).toMatchSnapshot()
  })
})
