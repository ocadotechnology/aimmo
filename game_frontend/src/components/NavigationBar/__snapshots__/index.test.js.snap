// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> renders correctly 1`] = `
<styled.nav>
  <WithStyles(AppBar)
    color="secondary"
    position="sticky"
  >
    <Styled(WithStyles(Toolbar))>
      <WithStyles(IconButton)
        aria-label="Close"
        color="inherit"
        href="/aimmo"
      >
        <CloseIcon />
      </WithStyles(IconButton)>
    </Styled(WithStyles(Toolbar))>
  </WithStyles(AppBar)>
</styled.nav>
`;

exports[`<NavigationBarLayout /> renders correctly 1`] = `
.c0 {
  grid-area: navigation-bar;
}

<nav
  className="c0"
/>
`;
